test-harness/tests/conftest.py

```diff
@@ -95,32 +95,16 @@ def syslog_mock(log_catcher):
 
 
 @pytest.fixture(scope='session')
-def dns_mock(log_catcher, navstar_ips, resolvconf_fixup):
-    """Set-up DNS mocks, both for agent AR (port 53) and master AR (port 61053)"""
-    m_61053 = DNSMock(log_catcher, port=61053)
-    m_61053.start()
+def dns_server_mock():
+    s = MesosLeaderDNSServer(
+        server_address=('127.0.0.1', 61053),
+        leader_ip='127.0.0.2',
+        )
+    s.start()
 
-    m_53 = DNSMock(log_catcher, port=53)
-    m_53.start()
+    yield s
 
-    yield (m_53, m_61053)
-
-    m_61053.stop()
-    m_53.stop()
-
-
-@pytest.fixture(scope='session')
-def navstar_ips():
```

Comment:
I've added back `navstar_ips` fixture and moved the dependency to the `nginx_class` fixture. One thing I don't understand though is what is being resolved with these IP addresses in case of `agent` AR instance. I'd expect some tests to fail if these were required.